### PR TITLE
Feature 349 request access

### PR DESCRIPTION
**Pull request by joester89**

### CODE REVIEW

**Comment by joester89** [comment]
@SamMousa although tests are in progress I would like a (quick) review 👼🏻

File: protected/assets/icons/svg/notification.svg [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (SamMousa) [inline protected/assets/icons/svg/notification.svg:1]: Why use an SVG icon? Where dit it come from?

File: protected/controllers/accessRequest/Respond.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        UserComponent $user,
+        AuthManager $abacManager,
+        AccessCheckInterface $accessCheck,
+        NotificationService $notificationService,
+        int $id
+    ) {
+        $accessRequest = AccessRequest::findOne(['id' => $id]);
+
+        $accessCheck->requirePermission($accessRequest, Permission::PERMISSION_RESPOND);
+        $model = \Yii::createObject(RespondFormModel::class, [
```
Comment (SamMousa) [inline protected/controllers/accessRequest/Respond.php:30]: Don't use `Yii::createObject()`. Either create the object directly or create a factory and inject it.

File: protected/jobHandlers/accessRequests/AccessRequestHandler.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<?php
+declare(strict_types=1);
+
+namespace prime\jobHandlers\accessRequests;
+
+use JCIT\jobqueue\exceptions\PermanentException;
+use JCIT\jobqueue\interfaces\JobHandlerInterface;
+use prime\models\ar\AccessRequest;
+
+abstract class AccessRequestHandler implements JobHandlerInterface
```
Comment (SamMousa) [inline protected/jobHandlers/accessRequests/AccessRequestHandler.php:10]: This should not be an abstract class, it should be a repository.

File: protected/jobHandlers/accessRequests/CreatedNotificationHandler.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+use JCIT\jobqueue\interfaces\JobInterface;
+use prime\jobs\accessRequests\ResponseNotificationJob;
+use prime\models\ar\AccessRequest;
+use prime\models\ar\User;
+use yii\mail\MailerInterface;
+use function iter\rewindable\map;
+use function iter\toArray;
+
+class CreatedNotificationHandler extends AccessRequestHandler
```
Comment (SamMousa) [inline protected/jobHandlers/accessRequests/CreatedNotificationHandler.php:14]: This should not extend here; this is a class hierarchy for the purpose of code reuse.
Use a repository that is injected (best imo), or a trait.

File: protected/jobHandlers/accessRequests/ImplicitlyGrantedHandler.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
+<?php
+declare(strict_types=1);
+
+namespace prime\jobHandlers\accessRequests;
+
+use JCIT\jobqueue\interfaces\JobInterface;
+use prime\jobs\accessRequests\ImplicitlyGrantedJob;
+use yii\mail\MailerInterface;
+
+class ImplicitlyGrantedHandler extends AccessRequestHandler
```
Comment (SamMousa) [inline protected/jobHandlers/accessRequests/ImplicitlyGrantedHandler.php:10]: See previous

File: protected/assets/icons/svg/notification.svg (hunk 1 above)
Comment (joester89) [reply protected/assets/icons/svg/notification.svg:1]: That should have been removed...

**Pull request closed without merging**